tokens/lib/Crypto/AnchorToken.hs

```diff
@@ -93,39 +119,29 @@ verifyToken
     -> m (Either String AnchorToken)
 verifyToken k txt = liftIO $ do
     now <- getCurrentTime
-
-    return . (B64.decode . encodeUtf8
-              >=> note "Bad signature" . getPayload k
-              >=> eitherDecodeStrict
-              >=> checkExpiry now) $ txt
+    return $ do
+        SignedAnchorToken sig payload <-
+            note "Failed to base64 decode" $ preview signedTokenByteString txt
+        note "Bad signature" (getPayload k sig payload)
+          >>= eitherDecodeStrict
+          >>= checkExpiry now
   where
     checkExpiry now tok
         | now > tok ^. tokenExpires = Left "Token expired"
         | otherwise                 = Right tok
 
 
--- | Prism between ByteStrings and tokens, signing and encoding one way,
--- checking signature and decoding the other.
---
--- NOTE: This does not check the token's expiry! Do /not/ use this to verify a
--- token.
+-- | Sign a token, to get something back out use verifyToken
 --
--- Encoding:
---
---    review signedBlob token
---
--- Decoding:
---
---    preview signedBlob token
---
-signed
+-- You'll need a key pair for this.
+signToken
     :: AnchorCryptoState Pair
-    -> Prism' Text AnchorToken
-signed pair = prism' enc dec
-  where
-    enc = decodeUtf8 . B64.encode . signPayload pair
-          . toStrict .  encode . toJSON
-    dec = hush . B64.decode . encodeUtf8 >=> getPayload pair >=> decodeStrict
+    -> AnchorToken
+    -> Text
```

Comment:
If this is a token ready for stuffing in an HTTP header, this could be a ByteString?